package.json

```diff
@@ -78,8 +78,8 @@
         "semver": "^7.3.5",
         "ts-loader": "^9.3.1",
         "ts-node": "^3.3.0",
-        "typescript": "^4.5.5",
-        "typescript4": "npm:typescript@~4.0.0",
+        "typescript": "^5.0.0",
```

Comment:
Hi @fseitun, we can't bump TypeScript across major versions without doing a new major version ourselves, which we're not ready to do at this time. The only way we would support TypeScript 5 is if we can support both 4 and 5 at the same time, which I honestly just haven't tested yet.

Here's how you would do it:
1. Instead of replacing `typescript4` in this file, add another line for `typescript5`
2. In the test file, add '5' to the array instead of replacing '4'

Then it's a matter of figuring out how to get the test for both 4 and 5 passing at the same time, which I anticipate is the actually interesting part of this.